internal/sdkprovider/service/vpc/aws_org_vpc_peering_connection.go

```diff
@@ -0,0 +1,211 @@
+		ForceNew:    true,
+		Description: "Identifier of the organization VPC.",
+	},
+	"aws_account_id": {
+		ForceNew:    true,
+		Required:    true,
+		Type:        schema.TypeString,
+		Description: userconfig.Desc("AWS account ID.").ForceNew().Build(),
+	},
+	"aws_vpc_id": {
+		ForceNew:    true,
+		Required:    true,
+		Type:        schema.TypeString,
+		Description: userconfig.Desc("AWS VPC ID.").ForceNew().Build(),
+	},
+	"aws_vpc_region": {
+		ForceNew:    true,
+		Required:    true,
+		Type:        schema.TypeString,
+		Description: userconfig.Desc("The AWS region of the peered VPC. For example, `eu-central-1`.").Build(),
+	},
+	"peering_connection_id": {
+		Computed:    true,
+		Type:        schema.TypeString,
+		Description: userconfig.Desc("The ID of the peering connection.").Build(),
+	},
+	"aws_vpc_peering_connection_id": {
+		Computed:    true,
+		Type:        schema.TypeString,
+		Description: "The ID of the AWS VPC peering connection.",
+	},
+	"state": {
+		Computed:    true,
+		Type:        schema.TypeString,
+		Description: userconfig.Desc("State of the peering connection.").PossibleValuesString(organizationvpc.VpcPeeringConnectionStateTypeChoices()...).Build(),
+	},
+}
+
+func ResourceAWSOrgVPCPeeringConnection() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Creates and manages an AWS VPC peering connection with an Aiven Organization VPC.",
+		CreateContext: common.WithGenClientDiag(resourceAWSOrgVPCPeeringConnectionCreate),
+		ReadContext:   common.WithGenClientDiag(resourceAWSOrgVPCPeeringConnectionRead),
+		DeleteContext: common.WithGenClientDiag(resourceAWSOrgVPCPeeringConnectionDelete),
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Timeouts: schemautil.DefaultResourceTimeouts(),
+
+		Schema: aivenAWSOrgVPCPeeringConnectionSchema,
+	}
+}
+
+func resourceAWSOrgVPCPeeringConnectionCreate(ctx context.Context, d *schema.ResourceData, client avngen.Client) diag.Diagnostics {
+	var (
+		orgID        = d.Get("organization_id").(string)
+		vpcID        = d.Get("organization_vpc_id").(string)
+		awsAccountID = d.Get("aws_account_id").(string)
+		awsVPCId     = d.Get("aws_vpc_id").(string)
+		awsRegion    = d.Get("aws_vpc_region").(string)
+
+		req = organizationvpc.OrganizationVpcPeeringConnectionCreateIn{
+			PeerRegion:       util.ToPtr(awsRegion),
+			PeerVpc:          awsVPCId,
+			PeerCloudAccount: awsAccountID,
+		}
+	)
+
+	pCon, err := createPeeringConnection(ctx, orgID, vpcID, client, d, req)
+	if err != nil {
+		return diag.Errorf("Error creating VPC peering connection: %s", err)
+	}
+
+	diags := getDiagnosticsFromState(newOrganizationVPCPeeringState(pCon))
+
+	d.SetId(schemautil.BuildResourceID(orgID, vpcID, awsAccountID, awsVPCId, awsRegion))
+
+	// in case of an error delete VPC peering connection
+	if diags.HasError() {
+		deleteDiags := resourceAzureOrgVPCPeeringConnectionDelete(ctx, d, client)
+		d.SetId("") // Clear the ID after delete
```

Comment:
nit: I wouldn't clear id if it wasn't deleted. Better keep the state to try again the deletion.